csharp/src/Ice/Communicator.cs

```diff
@@ -1068,54 +1027,53 @@ async Task PerformDisposeAsync()
 
                 if (_adminAdapter != null)
                 {
-                    Debug.Assert(_adminIdentity != null);
-                    return _adminAdapter.CreateProxy(_adminIdentity.Value, IObjectPrx.Factory);
+                    return _adminAdapter.CreateProxy(_adminIdentity!.Value, IObjectPrx.Factory);
```

Comment:
GetAdminAsync might return the proxy right away even though the get admin task isn't completed. I find it a bit strange that in one case GetAdminAsync will wait for full activation of the adapter and in another case, it doesn't. It would be better if GetAdminAsync always returned either as soon as the OA is available (but not necessarily activated) or only once it's full activated.

Reply:
Fixed by checking/reusing first _getAdminTask.